scripts/generate_maps.py

```diff
@@ -0,0 +1,177 @@
+import geopandas as gpd
+import pandas as pd
+import os
+import fiona
+from matplotlib_scalebar.scalebar import ScaleBar
+import matplotlib.pyplot as plt
+from matplotlib.patches import Patch
+
+# Define the paths
+input_file_path = "./antarctic_index.gpkg"
+map_path = "/Users/nathanbekele/Downloads/Quantarctica3/Miscellaneous/SimpleBasemap/ADD_DerivedLowresBasemap.shp"
+output_folder = "./data3"
+statistics_folder = "./data_statistics"
+
+# Ensure the output folders exist
+os.makedirs(output_folder, exist_ok=True)
+os.makedirs(statistics_folder, exist_ok=True)
+
+# Read the GeoPackage file into a GeoDataFrame
+fp = gpd.read_file(input_file_path)
+
+# Read the map shapefile
+mp = gpd.read_file(map_path)
+
+# Check if the 'Category' column exists
+if 'Category' in mp.columns:
+    # Define the colors for the categories
+    category_colors = {
+        'Ocean': '#a3bdd1',
+        'Ice shelf': '#cfe1eb',
+        'Land': '#f0f0f0',
+        'Sub-antarctic_G': 'lightgreen',
+        'Sub-antarctic_L': 'lightblue',
+        'Ice tongue': 'lightgrey',
+        'Rumple': 'yellow',
+    }
+
+    # Create a color column based on the 'Category' column
+    mp['color'] = mp['Category'].map(category_colors).fillna('grey')  # Default to grey for undefined categories
+
+    # Function to plot data on the colored basemap
+    def plot_data_on_basemap(basemap, gdf, institution, filename):
```

Comment:
It is odd to define a function like this inside of your `if` statement. Can you move it out to the top level?